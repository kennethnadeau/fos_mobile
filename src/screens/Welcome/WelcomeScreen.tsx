import React from "react";
import { View, StyleSheet } from "react-native";
import { Text } from "react-native-elements";
import { ScreenFC } from "react-native-navigation-register-screens";
import { SCREENS } from "@fos/constants";
import RabbitSvg from "@fos/assets/svg/rabbit.svg";
import { s, vs } from "react-native-size-matters";
import { Colors } from "themes";
import { Trans } from "react-i18next";

type WelcomeScreenProps = {
  fullName: string;
};

const WelcomeScreen: ScreenFC<WelcomeScreenProps> = ({ fullName }) => (
  <View style={styles.container}>
    <RabbitSvg
      color={Colors.secondary}
      fill={Colors.secondary}
      height={vs(74)}
      width={s(115)}
    />
    <Text style={styles.welcomeText}>
      <Trans values={{ fullName }}>Welcome Back</Trans>
    </Text>
  </View>
);

WelcomeScreen.screenName = SCREENS.WELCOME;

const styles = StyleSheet.create({
  container: {
    alignItems: "center",
    flex: 1,
    justifyContent: "center",
  },
  welcomeText: {
    fontSize: s(28),
    marginTop: vs(20),
    textAlign: "center",
    width: s(250),
  },
});

export default WelcomeScreen;
