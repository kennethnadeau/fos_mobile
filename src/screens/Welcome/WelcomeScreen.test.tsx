import React from "react";
import WelcomeScreen from "./WelcomeScreen";
import { render, RenderAPI } from "react-native-testing-library";

describe("<WelcomeScreen />", () => {
  let welcomeScreen: RenderAPI;

  beforeEach(() => {
    welcomeScreen = render(
      <WelcomeScreen componentId="id" fullName="John Doe" />,
    );
  });

  it("should show welcome back text", () => {
    const actual = welcomeScreen.queryByText("Welcome Back");
    expect(actual).not.toBeNull();
  });
});
