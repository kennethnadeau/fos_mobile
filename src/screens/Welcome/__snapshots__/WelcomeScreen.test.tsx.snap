// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WelcomeScreen /> should match snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <SvgMock
    color="#4CA1D6"
    fill="#4CA1D6"
    height={145.1705882352941}
    width={246.42857142857142}
  />
  <Text
    style={
      Object {
        "fontSize": 60,
        "marginTop": 39.23529411764706,
        "textAlign": "center",
        "width": 535.7142857142857,
      }
    }
    theme={
      Object {
        "colors": Object {
          "disabled": "hsl(208, 8%, 90%)",
          "divider": "#bcbbc1",
          "error": "#ff190c",
          "grey0": "#393e42",
          "grey1": "#43484d",
          "grey2": "#5e6977",
          "grey3": "#86939e",
          "grey4": "#bdc6cf",
          "grey5": "#e1e8ee",
          "greyOutline": "#bbb",
          "platform": Object {
            "android": Object {
              "error": "#f44336",
              "primary": "#2196f3",
              "secondary": "#9C27B0",
              "success": "#4caf50",
              "warning": "#ffeb3b",
            },
            "ios": Object {
              "error": "#ff3b30",
              "primary": "#007aff",
              "secondary": "#5856d6",
              "success": "#4cd964",
              "warning": "#ffcc00",
            },
          },
          "primary": "#2089dc",
          "searchBg": "#303337",
          "secondary": "#8F0CE8",
          "success": "#52c41a",
          "warning": "#faad14",
        },
      }
    }
  >
    Welcome Back
  </Text>
</View>
`;
